### PR TITLE
Security Privacy Remember me feature

### PR DESCRIPTION
**Pull request by SamuelSalas**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**
Pr covers the Remember Me feature from the Security and Privacy settings.

**Screenshots/Recordings**
https://app-automate.browserstack.com/builds/af352c3cb4fc61986127b52d3b21c37a2213b608/sessions/78f547e97075c8a784a408ff08e971e9009ee987?auth_token=cd78ef8aca878050fbc3c9234574b54af653db16eb91305f4f8080bd5a65b9c6

https://app-automate.browserstack.com/builds/af352c3cb4fc61986127b52d3b21c37a2213b608/sessions/bb44478ca30982be991cd948c05a6288f9f507d9?auth_token=03d4da1f2fe3e5ed386639b2f2904c0744d1aaa3a08886eb456b773ce7ed7085

https://app-automate.browserstack.com/dashboard/v2/public-build/cmw4WVBJcUxyaTkzOGhvWmJiWXNzeXFPcHlvdkJrNDI3ZWxEZHE2dEZiMXd3R01rTVJJNTB6TnRlSnVmdWg3SEpsSDRzZ3puNUUrUzk5Mml2c2ZrR2c9PS0tUkpCVytSaW1zSlEyS0hMMENUZUltUT09--4dc58e64fb259fdc3155aed09a0d4a7ea052e0be

Detox test results

image

1 deeplinks spec failed due to known ongoing issue.

**Issue**

Progresses #???

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/components/UI/SecurityOptionToggle/SecurityOptionToggle.tsx [hunk 1]
```diff
@@ -51,6 +52,7 @@ const SecurityOptionToggle = ({
           style={styles.switch}
           ios_backgroundColor={colors.border.muted}
           disabled={disabled}
+          {...generateTestId(Platform, testId)}
```
Comment (Andepande) [inline app/components/UI/SecurityOptionToggle/SecurityOptionToggle.tsx:55]: is testId a const?

**Review by Andepande — APPROVED** [review]

File: app/components/UI/SecurityOptionToggle/SecurityOptionToggle.tsx (hunk 1 above)
Comment (SamuelSalas) [reply app/components/UI/SecurityOptionToggle/SecurityOptionToggle.tsx:55]: Yes, but the value of testId is being pass a prop in SecurityOptionsToggleProps, line 21 to 28.

File: wdio/screen-objects/testIDs/Screens/LoginScreen.testIds.js [hunk 2]
```diff
@@ -7,3 +7,5 @@ export const LOGIN_VIEW_UNLOCK_BUTTON_ID = 'log-in-button';
 export const LOGIN_VIEW_SCREEN_ID = 'login';
 
 export const LOGIN_VIEW_TITLE_ID = 'login-title';
+
+export const LOGIN_WITH_REMEMBER_ME_SWITCH = 'login-with-remember-me-switch';
```
Comment (cortisiko) [inline wdio/screen-objects/testIDs/Screens/LoginScreen.testIds.js:11]: update detox. In particular, ensure that the [LoginView.js ](https://github.com/MetaMask/metamask-mobile/blob/main/e2e/pages/LoginView.js#L2) in the e2e folder is importing these constants.

File: wdio/screen-objects/testIDs/Screens/SecurityPrivacy.testIds.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export const SECURITY_PRIVACY_REMEMBER_ME_TOGGLE = 'turn-on-remember-me';
```
Comment (cortisiko) [inline wdio/screen-objects/testIDs/Screens/SecurityPrivacy.testIds.js:1]: update detox. In particular, ensure that the [SecurityAndPrivacyView.js](https://github.com/MetaMask/metamask-mobile/blob/main/e2e/pages/Drawer/Settings/SecurityAndPrivacy/SecurityAndPrivacyView.js#L7) in the e2e folder is importing these constants.

**Review by cortisiko — CHANGES_REQUESTED** [review]
code looks good. Left 2 comments. After you address these comments, rerun detox.

**Comment by SamuelSalas** [comment]
Most recent browserstack test run. https://app-automate.browserstack.com/builds/16af50c5bf7fb3eee3ac2c2a1b30550a68e3fef4/sessions/110ff9a64f6520ae8bd4d4ee74508aeb3b6544bd?auth_token=9231e4536c3f29b916e0ad55bff038ebddd30428488cfdcc1a61316d6fb31f63

**Review by cortisiko — CHANGES_REQUESTED** [review]

**Review by cortisiko — APPROVED** [review]
🌮 🌮 🌮 🌮

**Review by Andepande — APPROVED** [review]

**Pull request merged**